ui/nuxeo-document-comments/nuxeo-document-comment-thread.js

```diff
@@ -0,0 +1,311 @@
+              new CustomEvent('notify', {
+                composed: true,
+                bubbles: true,
+                detail: { message: this._computeTextLabel(this.level, 'fetch.error') },
+              }),
+            );
+            throw error;
+          }
+        });
+    }
+
+    _getCommentIndexById(commentId) {
+      return this.comments.findIndex((entry) => entry.id === commentId);
+    }
+
+    _handleCommentsChange(event) {
+      if (event.detail.path === 'comments.length') {
+        this.dispatchEvent(
+          new CustomEvent('number-of-replies', {
+            composed: true,
+            bubbles: true,
+            detail: { total: this.comments.length },
+          }),
+        );
+      }
+    }
+
+    _handleDeleteEvent(event) {
+      const index = this._getCommentIndexById(event.detail.commentId);
+      if (index !== -1) {
+        this.splice('comments', index, 1);
+        this._setTotal(this.total - 1);
+      }
+      event.stopPropagation();
+    }
+
+    _handleEditEvent(event) {
+      const index = this._getCommentIndexById(event.detail.commentId);
+      if (index !== -1) {
+        this.set(`comments.${index}.modificationDate`, event.detail.modificationDate);
+        this.set(`comments.${index}.text`, event.detail.text);
+      }
+      event.stopPropagation();
+    }
+
+    _loadMore() {
+      this._fetchComments(true);
+    }
+
+    _refresh() {
+      this.set('comments', []);
+      this._fetchComments(this.allCommentsLoaded);
+    }
+
+    _submitComment(e) {
+      if (e) {
+        e.preventDefault();
+      }
+      this._clearRequest();
+      this.$.commentRequest.data = {
+        'entity-type': 'comment',
+        parentId: this.uid,
+        text: this.text.trim(),
+      };
+
+      this.$.commentRequest
+        .post()
+        .then((response) => {
+          this._clearInput();
+          this.push('comments', response);
+          this._setTotal(this.total + 1);
+        })
+        .catch((error) => {
+          if (error.status === 404) {
+            this.dispatchEvent(
+              new CustomEvent('notify', {
+                composed: true,
+                bubbles: true,
+                detail: { message: this._computeTextLabel(this.level, 'notFound') },
+              }),
+            );
+          } else {
+            this.dispatchEvent(
+              new CustomEvent('notify', {
+                composed: true,
+                bubbles: true,
+                detail: { message: this._computeTextLabel(this.level, 'creation.error') },
+              }),
+            );
+            throw error;
+          }
+        });
+    }
+
+    /** Visibility Methods * */
+    _allowReplies(level) {
+      return level <= 2;
+    }
+
+    _moreAvailable(length, total, allCommentsLoaded) {
+      return length < total && !allCommentsLoaded;
+    }
+  }
+
+  customElements.define(DocumentCommentThread.is, DocumentCommentThread);
+  Nuxeo.DocumentComment = DocumentCommentThread;
```

Comment:
It should be:
```suggestion
  Nuxeo.DocumentCommentThread = DocumentCommentThread;
```

But we don't need to add the elements' classes to the Nuxeo namespace, unless it's a behavior or a mixin we need to make public. Element classes can be retrieved via `customElements.get('nuxeo-document-comment')`.

(added this to our checklist)